src/main/java/edu/wpi/first/nativeutils/vendordeps/OptionalTypeAdapter.java

```diff
@@ -0,0 +1,57 @@
+// modified from https://stackoverflow.com/questions/12161366/how-to-serialize-optionalt-classes-with-gson
+package edu.wpi.first.nativeutils.vendordeps;
+
+import java.io.IOException;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.Optional;
+
+import com.google.gson.Gson;
+import com.google.gson.TypeAdapter;
+import com.google.gson.TypeAdapterFactory;
+import com.google.gson.reflect.TypeToken;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonToken;
+import com.google.gson.stream.JsonWriter;
+
+class OptionalTypeAdapter<E> extends TypeAdapter<Optional<E>> {
+    public static final TypeAdapterFactory FACTORY = new TypeAdapterFactory() {
+        @SuppressWarnings("unchecked")
+        @Override
+        public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> type) {
+            Class<T> rawType = (Class<T>) type.getRawType().getClass();
+            if (rawType != Optional.class) {
+                return null;
+            }
+            final ParameterizedType parameterizedType = (ParameterizedType) type.getType();
+            final Type actualType = parameterizedType.getActualTypeArguments()[0];
+            final TypeAdapter<?> adapter = gson.getAdapter(TypeToken.get(actualType));
+            return new OptionalTypeAdapter(adapter);
+        }
```

Comment:
This should allow us to parse Optional correctly but has a few warnings to do with unchecked casts

Reply:
I don’t think that is worth adding, for something we don’t really need anyway. Empty string is just fine and we can specify that.

Reply:
I'll open a new pr to default it to an empty string and close this one